cmd/compose/create.go

```diff
@@ -110,8 +109,10 @@ func (opts createOptions) GetTimeout() *time.Duration {
 func (opts createOptions) Apply(project *types.Project) {
 	if opts.Pull != "" {
 		for i, service := range project.Services {
-			service.PullPolicy = opts.Pull
-			project.Services[i] = service
+			if opts.Pull != "missing" || service.PullPolicy == "" {
```

Comment:
Instead of looking for the default _value_, it would be better to (for these);

- use the `flag.Changed()` to detect whether or not the flag was set by the user
- change `createOptions.Pull` to a pointer (`nil` means "no value set", in which case the default can be used)

```go
if flags.Changed("pull") {
	opts.Pull = ...
}
```

Doing that makes the option work as expected; setting the flag _even if it's the default_ value has a different meaning than "not setting the option"; because the user may _intentionally_ want to set the pull behavior back to the default (overriding the one in the compose file)

Reply:
Good catch, I applied the same logic as for the other flags